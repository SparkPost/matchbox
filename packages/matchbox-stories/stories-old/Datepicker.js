import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { StoryContainer } from './helpers';

import { Datepicker, Panel } from '../src';

storiesOf('Datepicker', module)
  .addDecorator((getStory) => (
    <StoryContainer>{ getStory() }</StoryContainer>
  ))
  .add('Default',
  withInfo('Refer to http://react-day-picker.js.org/ for usage. The only props overridden by Matchbox are classNames and navbarElement.')(
  () => (
    <Panel sectioned>
      <Datepicker
        numberOfMonths={2}
        enableOutsideDays
        fixedWeeks
        initialMonth={new Date(2017, 5)}
        selectedDays={
          {
            from: new Date(2017, 5, 5),
            to: new Date(2017, 5, 10),
          }
        }
        disabledDays={{ after: new Date() }}
        onDayClick={action('Day Clicked')}
      />
    </Panel>
  )));
