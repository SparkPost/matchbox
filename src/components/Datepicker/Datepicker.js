import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import DayPicker, { DateUtils } from 'react-day-picker';

import { Icon } from '../Icon';
import styles from './Datepicker.module.scss';

class Navbar extends Component {
  render() {
    const {
      showPreviousButton,
      showNextButton,
      onPreviousClick,
      onNextClick
    } = this.props;

    return (
      <div className={styles.Navbar}>
        <Icon
          name='ArrowLeft'
          size={21}
          onClick={() => onPreviousClick()}
          className={styles.Prev} />
        <Icon
          name='ArrowRight'
          size={21}
          onClick={() => onNextClick()}
          className={styles.Next} />
      </div>
    )
  }
}

class Datepicker extends Component {
  render() {
    const modifiers = {
      firstSelected: (day) => DateUtils.isSameDay(day, this.props.selectedDays.from),
      lastSelected: (day) => DateUtils.isSameDay(day, this.props.selectedDays.to),
    };
    const modifiersStyles = {
      firstSelected: {
        color: 'white',
        backgroundColor: '#2693c3',
        borderTopLeftRadius: '4px',
        borderBottomLeftRadius: '4px'
      },
      lastSelected: {
        color: 'white',
        backgroundColor: '#2693c3',
        borderTopRightRadius: '4px',
        borderBottomRightRadius: '4px'
      },
    };
    return (
      <DayPicker
        modifiers={modifiers}
        classNames={styles}
        modifiersStyles={modifiersStyles}
        navbarElement={Navbar}
        {...this.props} />
    )
  }
};

export default Datepicker;
