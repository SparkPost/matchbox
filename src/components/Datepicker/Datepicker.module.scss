@import '../../styles/config';

/* DayPicker styles */
.container {
  display: block;
}

.wrapper {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  position: relative;
  user-select: none;
  outline: none;
  flex-direction: row;
  padding: 0;
}

.interactionDisabled {}

.month {
  margin: 0;
  flex: 0 0 auto;
  flex-grow: 1;
  flex-basis: 0;

  & + .month {
    margin-left: rem(18)
  }
}

.caption {
  height: rem(33);
  padding-top: 0;
  font-weight: 600;
  font-size: rem(14);
  text-align: center;
}

.weekdays {
  position: relative;
}

.weekdaysRow {
  display: flex;
}

.weekday {
  flex: 0 0 auto;
  flex-grow: 1;
  flex-basis: 0;

  padding: rem(4) rem(11);
  font-size: rem(12);
  text-align: center;
  color: color(gray, 5);

  abbr {
    border: none;
    text-decoration: none;
  }
}

.body {
  position: relative;
}

.week {
  display: flex;
}

.day {
  flex: 0 0 auto;
  flex-grow: 1;
  flex-basis: 0;
  min-width: rem(24);

  padding: rem(3) rem(7);
  margin-bottom: -1px;
  margin-right: -1px;

  border: 1px solid color(gray, 9);
  font-size: rem(12);
  font-weight: 400;
  text-align: center;
  cursor: pointer;
  vertical-align: middle;
  transition: 0.15s;

  &:focus {
    outline: none;
  }

  &:hover:not(.disabled):not(.selected) {
    background: color(gray, 9);
  }

  &:active:not(.disabled) {
    box-shadow: 0 0 0 1px color(gray, 8);
  }
}

.footer {}

.today {
  color: color(gray, 1);
  font-weight: 600;
}

.selected {
  background: rgba(color(blue), 0.3);
  border: 1px solid rgba(color(blue), 0.4);
  z-index: 100;
  color: color(gray, 1);
  font-weight: 600;
}

.outside {
  color: color(gray, 8);
  pointer-events: none;

  &.selected {
    background: rgba(color(blue), 0.12) !important;
    color: color(gray, 7) !important;
    border: 1px solid rgba(color(blue), 0.15);
  }
}

.disabled {
  color: color(gray, 8);
  pointer-events: none;
}

// Custom Components
.Navbar {
  position: absolute;
  left: 0;
  right: 0;
}

.Prev, .Next {
  position: absolute;
  top: 0;
  display: inline-block;
  cursor: pointer;

  fill: color(gray, 5);
  transition: 0.15s;

  &:hover {
    fill: color(blue, dark);
  }
}

.Prev { left: rem(24); }
.Next { right: rem(24); }
