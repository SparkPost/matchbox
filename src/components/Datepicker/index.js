export { default as Datepicker } from './Datepicker';
