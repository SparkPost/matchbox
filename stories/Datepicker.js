import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { StoryContainer } from './helpers';

import { Datepicker, Panel } from '../src';

storiesOf('Datepicker', module)
  .addDecorator((getStory) => (
    <StoryContainer>{ getStory() }</StoryContainer>
  ))
  .addWithInfo('Default',
  'Refer to http://react-day-picker.js.org/ for usage. The only props overridden by Matchbox are classNames and navbarElement.',
  () => (
    <Panel>
      <Datepicker
        numberOfMonths={2}
        enableOutsideDays
        fixedWeeks
        initialMonth={new Date(2017, 5)}
        selectedDays={
          {
            from: new Date(2017, 5, 5),
            to: new Date(2017, 5, 10),
          }
        }
        disabledDays={{ after: new Date() }}
        onDayClick={action('Day Clicked')}
      />
    </Panel>
  ));
