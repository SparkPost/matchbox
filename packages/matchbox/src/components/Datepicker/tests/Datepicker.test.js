import React from 'react';
import Datepicker from '../Datepicker';
import { shallow } from 'enzyme';
import moment from 'moment';

describe('Datepicker', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      selectedDays: {
        from: '2018-06-13T14:28:58.549Z',
        to: '2018-06-14T14:28:58.549Z'

      },
      initialMonth: moment('2018-06-13T14:28:58.549Z').toDate()
    };

    wrapper = shallow(<Datepicker {...props} />);
  });

  it('renders correctly with default props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('passes through all props', () => {
    wrapper.setProps({ foo: 'bar', foo2: 'bar2' });
    expect(wrapper).toMatchSnapshot();
  });




});
