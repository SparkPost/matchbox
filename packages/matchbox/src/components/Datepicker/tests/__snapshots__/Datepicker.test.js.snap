// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker passes through all props 1`] = `
<DayPicker
  canChangeMonth={true}
  captionElement={
    <Caption
      classNames={
        Object {
          "body": "DayPicker-Body",
          "caption": "DayPicker-Caption",
          "container": "DayPicker",
          "day": "DayPicker-Day",
          "disabled": "disabled",
          "footer": "DayPicker-Footer",
          "interactionDisabled": "DayPicker--interactionDisabled",
          "month": "DayPicker-Month",
          "navBar": "DayPicker-NavBar",
          "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
          "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
          "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
          "outside": "outside",
          "selected": "selected",
          "today": "today",
          "todayButton": "DayPicker-TodayButton",
          "week": "DayPicker-Week",
          "weekNumber": "DayPicker-WeekNumber",
          "weekday": "DayPicker-Weekday",
          "weekdays": "DayPicker-Weekdays",
          "weekdaysRow": "DayPicker-WeekdaysRow",
          "wrapper": "DayPicker-wrapper",
        }
      }
    />
  }
  classNames={null}
  enableOutsideDays={false}
  fixedWeeks={false}
  foo="bar"
  foo2="bar2"
  initialMonth={2018-06-13T14:28:58.549Z}
  labels={
    Object {
      "nextMonth": "Next Month",
      "previousMonth": "Previous Month",
    }
  }
  locale="en"
  localeUtils={
    Object {
      "default": Object {
        "formatDay": [Function],
        "formatMonthTitle": [Function],
        "formatWeekdayLong": [Function],
        "formatWeekdayShort": [Function],
        "getFirstDayOfWeek": [Function],
        "getMonths": [Function],
      },
      "formatDay": [Function],
      "formatMonthTitle": [Function],
      "formatWeekdayLong": [Function],
      "formatWeekdayShort": [Function],
      "getFirstDayOfWeek": [Function],
      "getMonths": [Function],
    }
  }
  modifiers={
    Object {
      "firstSelected": [Function],
      "lastSelected": [Function],
    }
  }
  modifiersStyles={
    Object {
      "firstSelected": Object {
        "backgroundColor": "#2693c3",
        "borderBottomLeftRadius": "4px",
        "borderTopLeftRadius": "4px",
        "color": "white",
      },
      "lastSelected": Object {
        "backgroundColor": "#2693c3",
        "borderBottomRightRadius": "4px",
        "borderTopRightRadius": "4px",
        "color": "white",
      },
    }
  }
  navbarElement={[Function]}
  numberOfMonths={1}
  pagedNavigation={false}
  renderDay={[Function]}
  reverseMonths={false}
  selectedDays={
    Object {
      "from": "2018-06-13T14:28:58.549Z",
      "to": "2018-06-14T14:28:58.549Z",
    }
  }
  showWeekNumbers={false}
  tabIndex={0}
  weekdayElement={<Weekday />}
/>
`;

exports[`Datepicker renders correctly with default props 1`] = `
<DayPicker
  canChangeMonth={true}
  captionElement={
    <Caption
      classNames={
        Object {
          "body": "DayPicker-Body",
          "caption": "DayPicker-Caption",
          "container": "DayPicker",
          "day": "DayPicker-Day",
          "disabled": "disabled",
          "footer": "DayPicker-Footer",
          "interactionDisabled": "DayPicker--interactionDisabled",
          "month": "DayPicker-Month",
          "navBar": "DayPicker-NavBar",
          "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
          "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
          "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
          "outside": "outside",
          "selected": "selected",
          "today": "today",
          "todayButton": "DayPicker-TodayButton",
          "week": "DayPicker-Week",
          "weekNumber": "DayPicker-WeekNumber",
          "weekday": "DayPicker-Weekday",
          "weekdays": "DayPicker-Weekdays",
          "weekdaysRow": "DayPicker-WeekdaysRow",
          "wrapper": "DayPicker-wrapper",
        }
      }
    />
  }
  classNames={null}
  enableOutsideDays={false}
  fixedWeeks={false}
  initialMonth={2018-06-13T14:28:58.549Z}
  labels={
    Object {
      "nextMonth": "Next Month",
      "previousMonth": "Previous Month",
    }
  }
  locale="en"
  localeUtils={
    Object {
      "default": Object {
        "formatDay": [Function],
        "formatMonthTitle": [Function],
        "formatWeekdayLong": [Function],
        "formatWeekdayShort": [Function],
        "getFirstDayOfWeek": [Function],
        "getMonths": [Function],
      },
      "formatDay": [Function],
      "formatMonthTitle": [Function],
      "formatWeekdayLong": [Function],
      "formatWeekdayShort": [Function],
      "getFirstDayOfWeek": [Function],
      "getMonths": [Function],
    }
  }
  modifiers={
    Object {
      "firstSelected": [Function],
      "lastSelected": [Function],
    }
  }
  modifiersStyles={
    Object {
      "firstSelected": Object {
        "backgroundColor": "#2693c3",
        "borderBottomLeftRadius": "4px",
        "borderTopLeftRadius": "4px",
        "color": "white",
      },
      "lastSelected": Object {
        "backgroundColor": "#2693c3",
        "borderBottomRightRadius": "4px",
        "borderTopRightRadius": "4px",
        "color": "white",
      },
    }
  }
  navbarElement={[Function]}
  numberOfMonths={1}
  pagedNavigation={false}
  renderDay={[Function]}
  reverseMonths={false}
  selectedDays={
    Object {
      "from": "2018-06-13T14:28:58.549Z",
      "to": "2018-06-14T14:28:58.549Z",
    }
  }
  showWeekNumbers={false}
  tabIndex={0}
  weekdayElement={<Weekday />}
/>
`;
